python/kfserving/kfserver/kfserver.py

```diff
@@ -0,0 +1,124 @@
+import tornado.ioloop
+import tornado.web
+import argparse
+import os
+import logging
+import json
+
+DEFAULT_HTTP_PORT = 8080
+DEFAULT_GRPC_PORT = 8081
+DEFAULT_PROTOCOL = "tensorflow/serving/predict.proto"
+DEFAULT_MODEL_NAME = "default"
+DEFAULT_LOCAL_MODEL_DIR = "/tmp/model"
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--http_port', default=DEFAULT_HTTP_PORT,
+                    help='The HTTP Port listened to by the model server.')
+parser.add_argument('--grpc_port', default=DEFAULT_GRPC_PORT,
+                    help='The GRPC Port listened to by the model server.')
+parser.add_argument('--model_dir', required=True,
+                    help='A URI pointer to the model directory')
+parser.add_argument('--model_name', default=DEFAULT_MODEL_NAME,
+                    help='The name that the model is served under.')
+parser.add_argument("--local_model_dir", default=DEFAULT_LOCAL_MODEL_DIR,
+                    help="The local path to copy model_dir.")
+
+args = parser.parse_args()
+
+KFSERVER_LOGLEVEL = os.environ.get('KFSERVER_LOGLEVEL', 'INFO').upper()
+logging.basicConfig(level=KFSERVER_LOGLEVEL)
+
+
+class KFServer(object):
+    def __init__(self):
+        self.http_port = args.http_port
+        self.grpc_port = args.grpc_port
+        self.protocol = DEFAULT_PROTOCOL
+
+        self.model_name = args.model_name
+        self.model_dir = args.model_dir
+        self.local_model_dir = args.local_model_dir
+
+    def start(self, models):
```

Comment:
Is it 1:1 mapping between server and model> self.model_name gives that impression, but then start takes multiple models...

Reply:
KFServer follows TensorflowServing's API of the ability of host multiple models. KFServer/XGBoost implements the server as a singleton.

Reply:
I think we are trying to support a model server that can host different versions of model right? The model map makes me a bit confused since that can contain different model names, but KFServer is initialized with a single model name.

Reply:
I want the base server to support multi-models (like TFServing). The initial impl of XGBoostServer is just a singleton. I've cleaned this up in the code though.

Reply:
How does handling different versions of models fit with the current control plane spec and "gitops"? 

I'm not sure we want to obscure what is being deployed. My feeling is TfServing came from a more monolithic serving architecture which expected to handle multiple models and versions and updates to them but what we are doing is more nimble where we handle canarys and updates at a higher level. I think it might be confusing if you have a server handling multiple models and you are also doing canary updates at a higher level. Am I missing something here or do you just mean a single model with multiple named inputs?

Reply:
As discussed today, this is to enable model server multi-tenancy use cases. The KFServing interface will not need to change. We can enable this as an optimization in the future.